fluent-jdbc/src/main/java/org/codejargon/fluentjdbc/internal/mappers/ObjectMapper.java

```diff
@@ -94,7 +94,7 @@ private Object optionalOf(Field field, Object value) {
             return Optional.empty();
         } else {
             Class<?> typeOfField = typeOfField(field);
-            if (!typeOfField.isAssignableFrom(value.getClass())) {
+            if (!value.getClass().isAssignableFrom(typeOfField)) {
```

Comment:
I removed the bytebuffer support for now and introduce it later.
I think it needs a bit of special handling, not easily implementable
through the plugins (similarly to Optional handling for instance)

On Fri, Jul 8, 2016 at 9:20 PM, J.Zimmermann <notifications@github.com>
wrote:

> In
> fluent-jdbc/src/main/java/org/codejargon/fluentjdbc/internal/mappers/ObjectMapper.java
> <https://github.com/zsoltherpai/fluent-jdbc/pull/34#discussion_r70127132>:
>
> > @@ -94,7 +94,7 @@ private Object optionalOf(Field field, Object value) {
> >              return Optional.empty();
> >          } else {
> >              Class<?> typeOfField = typeOfField(field);
> > -            if (!typeOfField.isAssignableFrom(value.getClass())) {
> > +            if (!value.getClass().isAssignableFrom(typeOfField)) {
>
> this change was required for detecting a ByteBuffer instances like the
> package internal HeapByteBuffer or DirectByteBuffer. I will try to find a
> better way and for now please remove the whole ByteBuffer stuff. The byte[]
> still works without any dirty workaround. I'm a beginner in reading writing
> groovy and my next pull requests contains units test too :)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/zsoltherpai/fluent-jdbc/pull/34/files/9e3af1a3c9dd10bc325d1ba15e1feeb7c93cb7df..c3b216da5c110f27209e3820c11bd8186f46ab75#r70127132>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB72xd1gERE8u1D2NAlc8q3ZefnRPDoTks5qTqL1gaJpZM4JHkWs>
> .
>